### PR TITLE
dlp: canonize snippets for jobs, deid, metadata, and risk

### PR DESCRIPTION
**Pull request by LogicalShark**
Match https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md

- Separate files for imports within region tags
- Example function parameters
- Error handling

### CODE REVIEW

File: dlp/snippets/deid/date_shift.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// deidentifyDateShift shifts dates found in the input between lowerBoundDays and
+// upperBoundDays.
+func deidentifyDateShift(w io.Writer, client *dlp.Client, project string, lowerBoundDays, upperBoundDays int32, input string) error {
```
Comment (tbpg) [inline dlp/snippets/deid/date_shift.go]: Initialize the client in the sample.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#initialize-clients-and-services-in-every-sample

File: dlp/snippets/deid/deid_fpe.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	dlp "cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// deidentifyFPE deidentifies the input with FPE (Format Preserving Encryption).
+// keyFileName is the file name with the KMS wrapped key and cryptoKeyName is the
+// full KMS key resource name used to wrap the key. surrogateInfoType is an
+// optional identifier needed for reidentification. surrogateInfoType can be any
+// value not found in your input.
+func deidentifyFPE(w io.Writer, client *dlp.Client, project, input string, infoTypes []string, keyFileName, cryptoKeyName, surrogateInfoType string) error {
```
Comment (tbpg) [inline dlp/snippets/deid/deid_fpe.go]: Include sample argument values.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#function-arguments

File: dlp/snippets/deid/deid_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	"testing"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+var client *dlp.Client
+var projectID string
+
+func TestMain(m *testing.M) {
```
Comment (tbpg) [inline dlp/snippets/deid/deid_test.go]: Won't need this once functions initialize their own client. Also, tests should probably just use `tc.ProjectID`.

File: dlp/snippets/jobs/delete.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// deleteJob deletes the job with the given name.
+func deleteJob(w io.Writer, client *dlp.Client, jobName string) error {
+	req := &dlppb.DeleteDlpJobRequest{
+		Name: jobName,
+	}
+	err := client.DeleteDlpJob(context.Background(), req)
+	if err != nil {
```
Comment (tbpg) [inline dlp/snippets/jobs/delete.go]: Inline.

File: dlp/snippets/deid/date_shift.go (hunk 1 above)
Comment (LogicalShark) [reply dlp/snippets/deid/date_shift.go]: Done

File: dlp/snippets/deid/deid_fpe.go (hunk 2 above)
Comment (LogicalShark) [reply dlp/snippets/deid/deid_fpe.go]: I tried to copy arguments from the Python Node.JS samples on https://cloud.google.com/dlp/docs/deidentify-sensitive-data. Should I keep the explanations in the function comment?

File: dlp/snippets/deid/deid_test.go (hunk 3 above)
Comment (LogicalShark) [reply dlp/snippets/deid/deid_test.go]: Done

File: dlp/snippets/jobs/delete.go (hunk 4 above)
Comment (LogicalShark) [reply dlp/snippets/jobs/delete.go]: Done

File: dlp/snippets/jobs/delete.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// deleteJob deletes the job with the given name.
+func deleteJob(w io.Writer, jobName string) error {
```
Comment (tbpg) [inline dlp/snippets/jobs/delete.go]: Add example `jobName`.

File: dlp/snippets/jobs/delete.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	ctx := context.Background()
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
+	}
+	req := &dlppb.DeleteDlpJobRequest{
+		Name: jobName,
+	}
+	if err = client.DeleteDlpJob(context.Background(), req); err != nil {
```
Comment (tbpg) [inline dlp/snippets/jobs/delete.go]: Use `ctx` instead of `context.Background()`.

File: dlp/snippets/jobs/list.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"google.golang.org/api/iterator"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// listJobs lists jobs matching the given optional filter and optional jobType.
+func listJobs(w io.Writer, project, filter, jobType string) error {
```
Comment (tbpg) [inline dlp/snippets/jobs/list.go]: Document args (move up from below).

File: dlp/snippets/deid/date_shift.go [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// deidentifyDateShift shifts dates found in the input between lowerBoundDays and
+// upperBoundDays.
+func deidentifyDateShift(w io.Writer, project string, lowerBoundDays, upperBoundDays int32, input string) error {
```
Comment (tbpg) [inline dlp/snippets/deid/date_shift.go]: s/project/projectID/g (in all of the samples).

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#common-identifiers

File: dlp/snippets/deid/date_shift.go [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+	// input := "2016-01-10"
```
Comment (tbpg) [inline dlp/snippets/deid/date_shift.go]: Missing `project` (`projectID`) and probably want these in order.

Same with other samples.

File: dlp/snippets/metadata/infoTypes.go [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
+)
+
+// infoTypes returns the info types in the given language and matching the given filter.
+func infoTypes(w io.Writer, languageCode, filter string) error {
```
Comment (tbpg) [inline dlp/snippets/metadata/infoTypes.go]: Document args.

File: dlp/snippets/deid/date_shift.go (hunk 8 above)
Comment (LogicalShark) [reply dlp/snippets/deid/date_shift.go]: Done

File: dlp/snippets/deid/date_shift.go (hunk 9 above)
Comment (LogicalShark) [reply dlp/snippets/deid/date_shift.go]: Done

File: dlp/snippets/jobs/delete.go (hunk 5 above)
Comment (LogicalShark) [reply dlp/snippets/jobs/delete.go]: Done

File: dlp/snippets/jobs/delete.go (hunk 6 above)
Comment (LogicalShark) [reply dlp/snippets/jobs/delete.go]: Done for all files

File: dlp/snippets/jobs/list.go (hunk 7 above)
Comment (LogicalShark) [reply dlp/snippets/jobs/list.go]: Fixed

File: dlp/snippets/metadata/infoTypes.go (hunk 10 above)
Comment (LogicalShark) [reply dlp/snippets/metadata/infoTypes.go]: Done

File: dlp/snippets/inspect/inspect_text_file.go [hunk 11]
```diff
@@ -21,13 +21,13 @@ import (
...
 
-	"cloud.google.com/go/dlp/apiv2"
+	dlp "cloud.google.com/go/dlp/apiv2"
 	dlppb "google.golang.org/genproto/googleapis/privacy/dlp/v2"
 )
 
 // inspectTextFile inspects a text file at a given filePath, and prints results.
 func inspectTextFile(w io.Writer, projectID, filePath string) error {
-	// projectID := "my-project-id"
+	// projectID := "my-project-id-id"
```
Comment (tbpg) [inline dlp/snippets/inspect/inspect_text_file.go]: s/id-id/id/

File: dlp/snippets/deid/deid_fpe.go (hunk 2 above)
Comment (tbpg) [reply dlp/snippets/deid/deid_fpe.go]: Yes, I think the explanations are fine to keep.

File: dlp/snippets/inspect/inspect_text_file.go (hunk 11 above)
Comment (LogicalShark) [reply dlp/snippets/inspect/inspect_text_file.go]: Done

File: dlp/snippets/jobs/jobs_test.go [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+func TestListJobs(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := new(bytes.Buffer)
+	listJobs(buf, tc.ProjectID, "", "RISK_ANALYSIS_JOB")
+	s := buf.String()
+	if len(s) == 0 {
+		// Create job.
+		riskNumerical(ioutil.Discard, tc.ProjectID, "bigquery-public-data", "risk-topic", "risk-sub", "nhtsa_traffic_fatalities", "accident_2015", "state_number")
```
Comment (tbpg) [inline dlp/snippets/jobs/jobs_test.go]: `riskNumerical` is undefined here. This doesn't compile. See Kokoro failures.

Comment (LogicalShark) [reply dlp/snippets/jobs/jobs_test.go]: Copied riskNumerical code

**Pull request merged**